lib/WorkerProcess.php

```diff
@@ -31,9 +30,33 @@ protected function doStart(Console $console): \Generator {
         });
 
         $server = yield from Internal\bootServer($this->logger, $console);
-        if ($console->isArgDefined("socket-transfer")) {
-            \assert(\extension_loaded("sockets") && PHP_VERSION_ID > 70007);
-            yield $server->start([new CommandClient((string) $console->getArg("config")), "importServerSockets"]);
+        if (\extension_loaded("sockets") && PHP_VERSION_ID > 70007 && stripos(PHP_OS, "WIN") !== 0) {
+            // needs socket_export_stream() and no Windows
+            $config = (string) $console->getArg("config");
+            $getSocketTransferImporter = function () use ($config, &$socketTransfer) {
+                return $socketTransfer ?? $socketTransfer = [new CommandClient($config), "importServerSockets"];
+            };
+
+            if (in_array(strtolower(PHP_OS), ["darwin", "freebsd"])) { // the OS not supporting *round-robin* so_reuseport client distribution
```

Comment:
Is that really `PHP_OS` or rather `PHP_OS_FAMILY`?

Reply:
PHP_OS_FAMILY only has been introduced recently. For now PHP_OS must suffice.

Reply:
See https://github.com/symfony/polyfill-php72 for a polyfill.

Reply:
Also, OpenBSD actually supports round-robin so_reuseport for example. Even though it is listed in the same family as FreeBSD (bsd). So, PHP_OS is definitely correct here.